controllers/wizard/fileUpload/form/PKPSubmissionFilesUploadBaseForm.php

```diff
@@ -262,7 +263,7 @@ public function getRevisionSubmissionFilesSelection($user, $uploadedFile = null)
             }
             if (
                 ($submissionFile->getFileStage() == SubmissionFile::SUBMISSION_FILE_REVIEW_ATTACHMENT || $submissionFile->getFileStage() == SubmissionFile::SUBMISSION_FILE_REVIEW_FILE) &&
-                $stageAssignmentDao->getBySubmissionAndRoleIds($submissionFile->getData('submissionId'), [Role::ROLE_ID_AUTHOR], $this->getStageId(), $user->getId())
+                $stageAssignmentDao->getBySubmissionAndRoleIds($submissionFile->getData('submissionId'), [Role::ROLE_ID_AUTHOR], $this->getStageId(), $user->getId())->next()
             ) {
```

Comment:
This is fix, to make this condition working as intended. But not sure if thats necessary anymore. As the logic seemed that `getSubmissionFiles` would return ALL files for given review round. And than the exclusions made sense to not expose some of these files to the Author.   But if getSubmissionFiles is now returning just the same fileType, I don't think thats necessary.  But keeping the changes minimum due to my limited understanding of this area :-).